src/posit/connect/oauth/sessions.py

```diff
@@ -0,0 +1,98 @@
+"""OAuth session resources."""
+
+from typing import List, overload
+
+from ..resources import Resource, Resources
+
+
+class Session(Resource):
+    """OAuth session resource.
+
+    Attributes
+    ----------
+    id : str
+        The internal numeric identifier of this OAuth session.
+    guid : str
+        The unique identifier of this OAuth session which is used in REST API requests.
+    user_guid : str
+        The unique identifier of the user that is associatid with this OAuth session.
+    oauth_integration_guid : str
+        The unique identifier of the OAuth integration that is associated with this OAuth session.
+    has_refresh_token : bool
+        Indicates whether this OAuth session has a refresh token.
+    created_time : str
+        The timestamp (RFC3339) indicating when this OAuth session was created.
+    updated_time : str
+        The timestamp (RFC3339) indicating when this OAuth session was last updated.
+    """
+
+    # Properties
+
+    @property
+    def id(self) -> str:
+        return self.get("id")  # type: ignore
+
+    @property
+    def guid(self) -> str:
+        return self.get("guid")  # type: ignore
+
+    @property
+    def user_guid(self) -> str:
+        return self.get("user_guid")  # type: ignore
+
+    @property
+    def oauth_integration_guid(self) -> str:
+        return self.get("oauth_integration_guid")  # type: ignore
+
+    @property
+    def has_refresh_token(self) -> bool:
+        return self.get("has_refresh_token")  # type: ignore
+
+    @property
+    def created_time(self) -> str:
+        return self.get("created_time")  # type: ignore
+
+    @property
+    def updated_time(self) -> str:
+        return self.get("updated_time")  # type: ignore
+
+    # CRUD Methods
+
+    def delete(self) -> None:
+        path = f"v1/oauth/sessions/{self.guid}"
+        url = self.url + path
+        self.session.delete(url)
+
+
+class Sessions(Resources):
+    @overload
+    def find(
+        self,
+        *,
+        all: bool = False,
+    ) -> List[Session]: ...
+
+    @overload
+    def find(self, **kwargs) -> List[Session]: ...
+
+    def find(self, **kwargs) -> List[Session]:
+        url = self.params.url + "v1/oauth/sessions"
```

Comment:
The default value for all in overload signature 1 isn't represented here. You should add something like the following if the API requires a default value. 

If you are saying that "the API defaults this value to False", then change the overload signature to `all: Optional[bool] = ...`. A docstring explaining that this value defaults to False starting in version, whatever, is helpful too.

```suggestion
    def find(self, **kwargs) -> List[Session]:
        if 'all' not in kwargs:
            kwargs['all'] = False
        url = self.params.url + "v1/oauth/sessions"
```

Reply:
yeah, the second description is the right one. The API defaults the value to false if none are provided.